conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/DeadlineTagTranslator.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.undertow.runtime;
+
+import com.palantir.tracing.TagTranslator;
+import io.undertow.server.HttpServerExchange;
+import io.undertow.util.HttpString;
+
+/** A {@link TagTranslator} which populates metadata from deadlines headers. */
+enum DeadlineTagTranslator implements TagTranslator<HttpServerExchange> {
+    INSTANCE;
+
+    private static final HttpString EXPECT_WITHIN = HttpString.tryFromString("Expect-Within");
```

Comment:
Thoughts on making `DeadlinesHttpHeaders` public for use here [(link)](https://github.com/palantir/deadlines-java/blob/360523410e0b415c26ab70dbaf9d6f49a6e95d6e/deadlines/src/main/java/com/palantir/deadlines/DeadlinesHttpHeaders.java#L20)?

Reply:
sure, let's do it: https://github.com/palantir/deadlines-java/pull/101